packages/components/src/filters/advanced/date-filter.js

```diff
@@ -0,0 +1,199 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import interpolateComponents from 'interpolate-components';
+import { SelectControl } from '@wordpress/components';
+import { find, partial } from 'lodash';
+import classnames from 'classnames';
+import { __, _x } from '@wordpress/i18n';
+import moment from 'moment';
+
+/**
+ * WooCommerce dependencies
+ */
+import { isoDateFormat, toMoment } from '@woocommerce/date';
+
+/**
+ * Internal dependencies
+ */
+import DatePicker from '../../calendar/date-picker';
+
+const dateStringFormat = __( 'MMM D, YYYY', 'wc-admin' );
+const dateFormat = __( 'MM/DD/YYYY', 'wc-admin' );
+
+class DateFilter extends Component {
+	constructor( { filter } ) {
+		super( ...arguments );
+
+		const [ isoAfter, isoBefore ] = ( filter.value || '' ).split( ',' );
+		const after = isoAfter ? toMoment( isoDateFormat, isoAfter ) : null;
+		const before = isoBefore ? toMoment( isoDateFormat, isoBefore ) : null;
+
+		this.state = {
+			before,
+			beforeText: before ? before.format( dateFormat ) : '',
+			beforeError: null,
+			after,
+			afterText: after ? after.format( dateFormat ) : '',
+			afterError: null,
+		};
+
+		this.onSingleDateChange = this.onSingleDateChange.bind( this );
+		this.onRangeDateChange = this.onRangeDateChange.bind( this );
+	}
+
+	getBetweenString() {
+		return _x(
+			'{{after /}}{{span}}and{{/span}}{{before /}}',
+			'Date range inputs arranged on a single line',
+			'wc-admin'
+		);
+	}
+
+	getLegend( filter, config ) {
+		const date = moment(); // For now...
```

Comment:
What's preventing the legend text from being generated from the filter value(s)?

Also, note that we're keeping the `<legend>` static and rendering a separate screen reader text component - see: https://github.com/woocommerce/wc-admin/pull/1089#discussion_r241843856 & #1099